tests/functional/adapter/grants/base_grants.py

```diff
@@ -0,0 +1,67 @@
+import os
+import pytest
+import snowflake.connector
+from dbt.tests.adapter.grants.base_grants import BaseGrants as OrigBaseGrants
+
+
+class BaseGrantsSnowflakePatch:
+    """
+    Overides the adapter BaseGrants to use new adapter functions
+    for grants.
+    """
+
+    def assert_expected_grants_match_actual(self, project, relation_name, expected_grants):
+        adapter = project.adapter
+        actual_grants = self.get_grants_on_relation(project, relation_name)
+        expected_grants_std = adapter.standardize_grant_config(expected_grants)
+
+        # need a case-insensitive comparison -- this would not be true for all adapters
+        # so just a simple "assert expected == actual_grants" won't work
+        diff_a = adapter.diff_of_grants(actual_grants, expected_grants_std)
+        diff_b = adapter.diff_of_grants(expected_grants_std, actual_grants)
+        assert diff_a == diff_b == {}
+
+
+class BaseGrantsSnowflake(BaseGrantsSnowflakePatch, OrigBaseGrants):
+    @pytest.fixture(scope="session", autouse=True)
+    def ensure_database_roles(project):
```

Comment:
To allow tests that use database_roles, this fixture has been added at session level to create 3 database roles in a similar style to the account-level roles that exist in the testing framework.